lib/spack/spack/test/build_environment.py

```diff
@@ -47,25 +47,15 @@ def build_environment():
 
     yield {'cc': cc, 'cxx': cxx, 'fc': fc}
 
-    for name in ('SPACK_CC', 'SPACK_CXX', 'SPACK_FC', 'SPACK_PREFIX',
-                 'SPACK_ENV_PATH', 'SPACK_DEBUG_LOG_DIR',
-                 'SPACK_COMPILER_SPEC', 'SPACK_SHORT_SPEC',
-                 'SPACK_CC_RPATH_ARG', 'SPACK_CXX_RPATH_ARG',
-                 'SPACK_F77_RPATH_ARG', 'SPACK_FC_RPATH_ARG',
-                 'SPACK_SYSTEM_DIRS'):
-        del os.environ[name]
-
 
 def test_static_to_shared_library(build_environment):
     os.environ['SPACK_TEST_COMMAND'] = 'dump-args'
 
     expected = {
-        'linux': ('/bin/mycc -Wl,-rpath,/spack-test-prefix/lib'
-                  ' -Wl,-rpath,/spack-test-prefix/lib64 -shared'
+        'linux': ('/bin/mycc -shared'
```

Comment:
Why are we no longer emitting rpaths connecting each package to its own lib dir? This seems to go across all tests.

Reply:
The `cc` wrapper itself no longer handles adding `<prefix>/lib`, its only job is to add the lib directories that the `SPACK_LINK_DIRS` environment variable tells it to. `build_environment` is responsible for making sure that `SPACK_LINK_DIRS` contains the `lib`/`lib64` directories when appropriate.

Reply:
So the test for this should be covered by my request in https://github.com/spack/spack/pull/8136#discussion_r256120589?

Reply:
Yes